go/src/koding/db/models/data.go

```diff
@@ -0,0 +1,59 @@
+package models
+
+import (
+	"strings"
+)
+
+// Data holds arbitrary key value.
+type Data map[string]interface{}
+
+// GetString returns value as string with given key
+func (d *Data) GetString(key string) (string, error) {
+	dt, err := d.Get(key)
+	if err != nil {
+		return "", err
+	}
+
+	s, ok := dt.(string)
+	if !ok {
+		return "", ErrDataInvalidType
+	}
+
+	return s, nil
+}
+
+// Get returns value with given key
+func (d *Data) Get(key string) (interface{}, error) {
+	if d == nil {
+		return "", ErrDataKeyNotExists
+	}
+
+	dt, ok := findPath(strings.Split(key, "."), *d)
+	if !ok {
+		return "", ErrDataKeyNotExists
+	}
+
+	return dt, nil
+}
+
+func findPath(path []string, data Data) (interface{}, bool) {
```

Comment:
It could take ... string instead (`findPath(data Data, path ...string)`), so the API could be simpler:

```go
d, ok = findPath(testDataMap, nonExistingKey)
```

Instead of:

```go
d, ok = findPath([]string{nonExistingKey}, testDataMap)
```